pymc4/distributions/continuous.py

```diff
@@ -1174,6 +1177,84 @@ def upper_limit(self):
         return self.conditions["high"]
 
 
+class Flat(ContinuousDistribution):
+    r"""A uniform distribution with support :math:`(-\inf, \inf)`.
+    Used as a uninformative log-likelihood that returns
+    zeros regardless of the passed values.
+    """
+
+    def __init__(self, name, **kwargs):
+        super().__init__(name, **kwargs)
+
+    @staticmethod
+    def _init_distribution(conditions):
+        return tfd.Uniform(low=-np.inf, high=np.inf)
+
+    def log_prob(self, value):
+        # convert the value to tensor
+        value = tf.convert_to_tensor(value)
+        expected = tf.zeros(self._distribution.batch_shape + self._distribution.event_shape)
+        # check if the event shape matches
+        if (
+            len(self._distribution.event_shape)
```

Comment:
A minor nitpick, our distributions now have the `event_shape` and `batch_shape` properties, so you don't have to access the `_distribution` attribute to look them up. You can do `self.event_shape` and `self.batch_shape` instead